src/library/searchqueryparser.cpp

```diff
@@ -5,8 +5,9 @@
 const char* kNegatePrefix = "-";
 const char* kFuzzyPrefix = "~";
 
-SearchQueryParser::SearchQueryParser(QSqlDatabase& database)
```

Comment:
The test for the SearchQueryParser uses this constructor 
and I couldn't figure out a way to get rid of it.
If you see the code it connects it to a new instance of the DB. 
In the test there is no mention of a TrackCollection so SearchQueryParser 
has two constructors now. Thankfully the constructor is only 
called once in the whole codebase (at TrackCollection's constructor) 
so it's not a huge problem. But I don't like it. From what I understand 
it's normal to have functions that are used only for testing 
so my concern might be a bit off.